doc/styleguide.md

```diff
@@ -0,0 +1,31 @@
+# Styleguide
+
+Please refer to the *"official"* [scala style guide](https://docs.scala-lang.org/style/) @ and by scala-lang.org. This
+document will only note any style decisions made for this project that differ from the style descibed there as well as
+clarify where said guide leaves room for interpretation or offers alternatives.
+
+This document is also work-in-progress, so please feel free to create pull requests or issues for any open questions or
+suggestions.
+
+## Scaladoc
+
+Use scaladoc style with gutter asterisks aligned in column three:
+```scala
+/** Adds two numbers.
+  *
+  * Does a really good job at adding two [[Int]] numbers.
+  *
+  * @param  number        first summand
+  * @param  anotherNumber second summand
+  * @return the sum of the two numbers passed as parameters
+  */
+def customAdd(number: Int, anotherNumber: Int): Int = number + anotherNumber
+```
+
+Scaladoc checklist:
+* A short function/class/trait description starting in the very first line followed by two newlines.
+* An optional text block elaborating on the function/class/trait followed by two newlines.
+* If applicable: special notifications (`@note`), possible exceptions (`@throws`), parameters (`@param`), type
+  parameters (`@tparam`), return value (`@return`), etc.
+* The names - or types for e.g. `@throws` statements - and descriptions are tab-aligned if reasonable, i.e. the
```

Comment:
Can we use space-alignment? I've configured my IDE to convert `<tab>` to two spaces for scala files.
IMO it's always mixed if one allows tabs to be used.

Reply:
Uhm yeah, I exclusively use spaces as well. What I actually meant by tab alignment is having them start in the same column / tab-stop. I'll change the wording.